src/main/java/com/dm/dmservicesandbox/component/UserVerificationComponent.java

```diff
@@ -0,0 +1,126 @@
+package com.dm.dmservicesandbox.component;
+
+import com.dm.dmservicesandbox.dbhome.UserTokenRepository;
+import com.dm.dmservicesandbox.domain.EmailData;
+import com.dm.dmservicesandbox.domain.UserToken;
+import com.dm.dmservicesandbox.domain.ServiceReponse;
+
+import jakarta.persistence.EntityManager;
+
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Component;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.servlet.support.ServletUriComponentsBuilder;
+
+import java.security.SecureRandom;
+import java.math.BigInteger;
+import java.text.SimpleDateFormat;
+import java.util.Calendar;
+import java.util.Date;
+import java.util.TimeZone;
+
+
+@Component
+public class UserVerificationComponent {

```

Comment:
May we can call this as TokenProvider.java

Reply:
This component not only create the token, but also implements the entire verification process - save token to db before emailing the link; email the link; compare the token in the api and the token retrieved from db for verification; check if the token is expired; clear the token after verification is successfull.